fineract-provider/src/test/java/org/apache/fineract/infrastructure/core/LiquibaseDisabledScenario.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.BDDMockito.given;
+
+import com.zaxxer.hikari.HikariDataSource;
+import io.cucumber.java8.En;
+import org.apache.fineract.infrastructure.core.config.FineractProperties;
+import org.apache.fineract.infrastructure.core.service.migration.ExtendedSpringLiquibaseFactory;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDataSourceFactory;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDatabaseStateVerifier;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDatabaseUpgradeService;
+import org.apache.fineract.infrastructure.security.service.TenantDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class LiquibaseDisabledScenario implements En {
+
+    @Autowired
+    private TenantDetailsService tenantDetailsService;
+
+    @Autowired
+    private HikariDataSource tenantDataSource;
+
+    @Autowired
+    private FineractProperties fineractProperties;
+
+    @Autowired
+    private TenantDatabaseStateVerifier databaseStateVerifier;
+
+    @Autowired
+    private ExtendedSpringLiquibaseFactory liquibaseFactory;
+
+    @Autowired
+    private TenantDataSourceFactory tenantDataSourceFactory;
+
+    @Autowired
+    private TenantDatabaseUpgradeService underTest;
+
+    public LiquibaseDisabledScenario() {
+        Given("Liquibase is disabled", () -> {
+            given(databaseStateVerifier.isLiquibaseDisabled()).willReturn(true);
+        });
+
+        When("After properties set is executed", () -> {
```

Comment:
Wouldn't it make more sense to refer to a business event here instead of "afterpropertiesset"? I'm thinking about `When("The database migration process is executed")`

Reply:
Done